src/Types/Filtering.hs

```diff
@@ -38,11 +43,22 @@ data FunctionSignature =
   }
 
 instance Show FunctionSignature where
-  show (FunctionSignature constraints argsType returnType) = 
+  show (FunctionSignature constraints argsType returnType) =
     printf "(%s) => %s" constraintsExpr argsExpr
       where
         constraintsExpr = (intercalate ", " . map show) constraints
         argsExpr = (intercalate " -> " . map show) (argsType ++ [returnType])
 
-defaultTimeout = 3000000 :: Int
-defaultNumChecks = 10 :: Int
\ No newline at end of file
+data SampleResult = SampleResult
+  { _inputs :: [String], _results :: [[String]]}
```

Comment:
Only note from today was to change this type to something more descriptive (like a list of pairs, containing the body of the function this was run on)